node/pkg/governor/governor.go

```diff
@@ -669,15 +711,26 @@ func (gov *ChainGovernor) CheckPendingForTime(now time.Time) ([]*common.MessageP
 		gov.msgsToPublish = nil
 	}
 
-	for _, ce := range gov.chains {
+	// Iterate deterministically by accessing keys from this slice instead of the chainEntry map directly
+	for _, chainId := range gov.chainIds {
+		ce, ok := gov.chains[chainId]
+		if !ok {
```

Comment:
Since this happens on the processor routine, I think it would definitely be better to avoid these map look ups and replace `gov.chainIds` with a slice of pointers.

Reply:
I've added https://github.com/wormhole-foundation/wormhole/issues/4029 so we can discuss future performance enhancements

Reply:
#4029 recorded as an Issue here